plugins/interactive/interactive.go

```diff
@@ -0,0 +1,287 @@
+	"github.com/taskcluster/taskcluster-worker/plugins"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"github.com/taskcluster/taskcluster-worker/runtime/ioext"
+)
+
+// defaultArtifactPrefix is the default artifact prefix used if nothing is
+// configured or given in the task definition
+const defaultArtifactPrefix = "private/interactive/"
+
+// defaultShellToolURL is the default URL for the tool that can connect to the
+// shell socket and display an interactive shell session.
+const defaultShellToolURL = "https://tools.taskcluster.net/shell/"
+
+// defaultShellToolURL is the default URL for the tool that can list displays
+// and connect to the display socket with interactive noVNC session.
+const defaultDisplayToolURL = "https://tools.taskcluster.net/display/"
+
+type provider struct {
+	plugins.PluginProviderBase
+}
+
+func (provider) ConfigSchema() schematypes.Schema {
+	return configSchema
+}
+func (provider) NewPlugin(options plugins.PluginOptions) (plugins.Plugin, error) {
+	var c config
+	if schematypes.MustMap(configSchema, options.Config, &c) != nil {
+		return nil, engines.ErrContractViolation
+	}
+	if c.ArtifactPrefix == "" {
+		c.ArtifactPrefix = defaultArtifactPrefix
+	}
+	if c.ShellToolURL == "" {
+		c.ShellToolURL = defaultShellToolURL
+	}
+	if c.DisplayToolURL == "" {
+		c.DisplayToolURL = defaultDisplayToolURL
+	}
+	return &plugin{
+		config: c,
+		log:    options.Log,
+	}, nil
+}
+
+type plugin struct {
+	plugins.PluginBase
+	config config
+	log    *logrus.Entry
+}
+
+func (p *plugin) PayloadSchema() schematypes.Object {
+	s := schematypes.Object{
+		MetaData: schematypes.MetaData{
+			Title: "Interactive Features",
+			Description: `Settings for interactive features, all options are optional
+				an empty object can be used to enable the interactive features with
+				default options.`,
+		},
+		Properties: schematypes.Properties{
+			"disableDisplay": schematypes.Boolean{},
+			"disableShell":   schematypes.Boolean{},
+		},
+	}
+	if !p.config.ForbidCustomArtifactPrefix {
+		s.Properties["artifactPrefix"] = schematypes.String{
+			MetaData: schematypes.MetaData{
+				Title: "Artifact Prefix",
+				Description: "Prefix for the interactive artifacts will be used to " +
+					"create `<prefix>/shell.html`, `<prefix>/display.html` and " +
+					"`<prefix>/sockets.json`. The prefix defaults to `" +
+					p.config.ArtifactPrefix + "`",
+			},
+			Pattern:       `^[\x20-.0-\x7e][\x20-\x7e]*/$`,
+			MaximumLength: 255,
+		}
+	}
+	return schematypes.Object{
+		Properties: schematypes.Properties{
+			"interactive": s,
+		},
+	}
+}
+
+func (p *plugin) NewTaskPlugin(options plugins.TaskPluginOptions) (
+	plugins.TaskPlugin, error,
+) {
+	var P payload
+	if schematypes.MustMap(p.PayloadSchema(), options.Payload, &P) != nil {
+		return nil, engines.ErrContractViolation
+	}
+	// If not always enabled or no options are given then this is disabled
+	if P.Interactive == nil && !p.config.AlwaysEnabled {
+		return nil, nil
```

Comment:
it's allowed in the current plugin specification... We can also return `plugins.TaskPluginBase{}`

But allowing for nil seems okay, as we can optimize a bit by doing it... basically, we won't have a lot of stupid base plugins that does nothing to call..